docs/proposals/remediation/cluster-status-based-remedy.md

```diff
@@ -0,0 +1,379 @@
+
+Users expect that when the system detects a member cluster failure, it will automatically trigger cluster business traffic removal to ensure that business requests will not be load balanced to the failed cluster, causing service unavailability. In addition, member clusters need to upgrade the Kubernetes version. Services deployed on clusters that are in the upgrade process may be in an unavailable state. Users need to manually trigger cluster traffic removal to avoid service unavailability.
+
+### Goals
+
+- Ability to detect domain name functional anomalies in Kubernetes clusters.
+- Provide the function to automatically trigger the removal of all MultiClusterIngress business traffic on the kubernetes cluster.
+- Provide the function to manually trigger the removal of all MultiClusterIngress business traffic on the kubernetes cluster.
+
+### Non-Goals
+
+In this proposal, we only detect the domain name resolution function within the cluster.
+
+- Do not list all cluster statuses that need to be detected.
+- Not all actions that the remedy policy can trigger execution are listed.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+The administrator is preparing to upgrade the Kubernetes version of the member1 cluster. In order to prevent the data that may be caused by the cluster upgrade from affecting the business of users on the cluster, an immediate policy is created in the system to trigger the system to remove all business traffic on the member1 cluster to avoid End user access is affected.
+
+#### Story 2
+
+The administrator enables the capability by configuring remedy policies in the system: When the system detects an abnormality in the CoreNDS domain name resolution capability in the member1 cluster, the system automatically removes all MultiClusterIngress traffic on the member1 cluster to prevent end users from accessing the MultiClusterIngress backend through the LoadBalance service.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+集群内域名解析功能探测可能不够精确，当用户通过其他工具或者凭借运维经验发现集群内域名解析功能不可用时，也可以主动上报，将信息同步到Cluster对象的condition中。
+
+## Design Details
+
+### Cluster status management
+
+We do cluster status management by extending the ConditionType in the Cluster object. A new `ServiceDomainNameResolution` ConditionType is added to represent the domain name resolution function within the cluster.
+
+Q: Why not introduce a new API for cluster status management?
+A: Compared with using `Cluster` API, the newly introduced API is less convenient for users to understand and integrate with third-party systems; in addition, because it currently only detects the domain name resolution function in the cluster and there are few scene inputs, the new API designed may have insufficient scalability. For the current scenario, using Cluster objects for expansion meets the needs.
```

Comment:
Please move this part to `Alternatives` below.

Reply:
Updated, please take a look again.